dev/com.ibm.ws.security.audit.source/src/com/ibm/ws/security/audit/encryption/AuditCrypto.java

```diff
@@ -2109,6 +2113,68 @@ static final byte[] decrypt(byte[] mesg, byte[] key, String cipher, boolean useJ
         return tmpMesg;
     }
 
+    /**
+     * @param key
+     * @param cipher
+     * @return
+     * @throws InvalidKeyException
+     * @throws NoSuchAlgorithmException
+     * @throws InvalidKeySpecException
+     */
+    private static SecretKey constructSecretKey(byte[] key, String cipher) throws InvalidKeyException, NoSuchAlgorithmException, InvalidKeySpecException, NoSuchProviderException {
+        SecretKey sKey = null;
+        if (cipher.indexOf("AES") != -1) {
+            // 16 bytes = 128 bit key
+            sKey = new SecretKeySpec(key, 0, 16, "AES");
+        } else {
+            DESedeKeySpec kSpec = new DESedeKeySpec(key);
+            SecretKeyFactory kFact = null;
+
+            kFact = (provider == null) ? SecretKeyFactory.getInstance(encryptAlgorithm) : SecretKeyFactory.getInstance(encryptAlgorithm, provider);
+
+            sKey = kFact.generateSecret(kSpec);
+        }
+        return sKey;
+    }
+
+    /**
+     * @param key
+     * @param cipher
+     * @param sKey
+     * @return
+     * @throws NoSuchAlgorithmException
+     * @throws NoSuchPaddingException
+     * @throws InvalidKeyException
+     * @throws InvalidAlgorithmParameterException
+     */
+    private static Cipher createCipher(int cipherMode, byte[] key, String cipher,
+                                       SecretKey sKey) throws NoSuchAlgorithmException, NoSuchPaddingException, InvalidKeyException, InvalidAlgorithmParameterException, NoSuchProviderException {
+
+        Cipher ci = null;
+        ci = (provider == null) ? Cipher.getInstance(cipher) : Cipher.getInstance(cipher, provider);
+
+        if (cipher.indexOf("ECB") == -1) {
+            if (cipher.indexOf("GCM") != -1) {
+                byte[] iv = new byte[12];
```

Comment:
Does `iv` need to be populated with bytes from the `key` like in the other Ciphers?

Reply:
Yes, fixed it.